app/src/pages/outside/loginPage/pageBlocks/loginBlock/loginBlock.jsx

```diff
@@ -48,29 +50,17 @@ export class LoginBlock extends PureComponent {
     extensions: [],
   };
 
-  getLoginPageExtensions = () => {
+  getSignUpExtension = () => {
     const { extensions } = this.props;
 
-    return extensions.reduce(
-      (acc, extension) => ({
-        ...acc,
-        [extension.name]: {
-          name: extension.title || extension.name,
-          // TODO link generation
-          link: '/signUp',
-          component: <ExtensionLoader extension={extension} />,
-          mobileDisabled: true,
-        },
-      }),
-      {},
-    );
+    return extensions.find((extension) => extension.pluginName === SIGN_UP_PLUGIN_NAME);
   };
 
   render() {
     return (
       <PageBlockContainer header={messages.welcome} hint={messages.login}>
         <LoginForm />
-        {this.getLoginPageExtensions().signUp && this.getLoginPageExtensions().signUp.component}
+        {this.getSignUpExtension() && <ExtensionLoader extension={this.getSignUpExtension()} />}
```

Comment:
No need to hardcode this for sign up flow plugin as it is private feature.
You can just render the collection of extensions via `extensions.map`.